config/v1/types_infrastructure.go

```diff
@@ -229,6 +229,18 @@ const (
 	IBMCloudProviderTypeUPI IBMCloudProviderType = "UPI"
 )
 
+// UserProvidedDNS indicates the state of the DNS configuration.
+// +kubebuilder:validation:Enum="";Enabled;Disabled
```

Comment:
Do you need the empty string case?

Reply:
What happens during upgrades? If we use kubebuilder to default,
`// +kubebuilder:default:="Disabled"`, afaik this field will be "Disabled" after the upgrade to 4.15. In that case, we can remove "" from the enum add the default.

Reply:
@JoelSpeed It appears that I have to either get "" as an option or keep omitempty. I chose to keep omit empty for each entry. It was coming up with an error stating just that.